### PR TITLE
web: Tufte-style sidenotes and justified text

### PR DESCRIPTION
**Pull request by plt-amy**
Not much to say here, just gotta wait for the preview to render, I suppose.

### CODE REVIEW

**Comment by Lavenza** [comment]
[Pull request preview](https://preview.1lab.dev/338)

**Comment by ncfavier** [comment]
The sidebar is gone for me in the preview :(

**Comment by SquidDev** [comment]
Is that another instance of #147, just now more visible as the breakpoints have changed again?

File: support/web/js/lib/settings.tsx [hunk 1]
```diff
@@ -51,9 +52,12 @@ export type Theme = 'light' | 'dark' | 'system';
 export const equationSetting   = new Setting<boolean>("equations",   false)
   .onChange((v) => v ? clz.add("show-equations") : clz.remove("show-equations"));
 
-export const serifFontSetting  = new Setting<boolean>("sans_serif",  false)
+export const serifFontSetting  = new Setting<boolean>("sans_serif", false)
   .onChange((v) => v ? clz.remove("sans-serif") : clz.add("sans-serif"));
```
Comment (SquidDev) [inline support/web/js/lib/settings.tsx:56]: Wait, is this option the wrong way round? When `sans_serif` is false, we add the `sans-serif` font!

I'm not sure what the intended default is, but we should possibly update the CSS to match it - currently that prefers serif fonts over sans serif ones when JS is disabled.

**Review by SquidDev — APPROVED** [review]
This is really nice! It's depressing we need to rely on JS at all here, I'm amazed how well it works!

The only problem I've noticed is that footnotes aren't visible when JS is disabled. We could fix it by adding a CSS class to the body from Javascript to enable the footnotes, but I'm not sure either of us can face wrangling the CSS for that, so inclined to say ship it as is :D.

File: support/web/css/vars.scss [hunk 2]
```diff
@@ -105,3 +105,6 @@ $light-button-hover-bg: $zinc-200;
 
 $dark-text-bg: #23272E;
 $dark-text-fg: #ABB2BF;
+
+// "Desktop layout": navigation bar, content, sidenotes.
+$desktop-layout-width: 110rem;
```
Comment (ncfavier) [inline support/web/css/vars.scss:110]: Given https://github.com/plt-amy/1lab/issues/147, we should keep this low. The minimum for my 1920x1080, 120 DPI laptop screen seems to be 1920*96/120/16 = 96rem, minus one for safety:

```suggestion
$desktop-layout-width: 95rem;
```

**Comment by ncfavier** [comment]
The targets of footnote links (like `#fn1`) are now missing, so clicking them doesn't do anything. Is that intended?

**Comment by plt-amy** [comment]
@ncfavier Yeah, it's intentional. Since browsers won't move you to a hash that's not painted, clicking the footnote references on the desktop layout does nothing. But when the screen is narrow and the footnotes render again, the footnote references just work, without us having to shuffle the DOM, or override `click` for the anchors, etc.

**Review by Lavenza — APPROVED** [review]

File: support/web/css/code.scss [hunk 3]
```diff
@@ -13,110 +12,115 @@
...
 }
 
-code, pre, .sourceCode {
-  font-family: 'Julia Mono';
-  font-size: var(--code-font-size);
-  font-weight: 500;
-}
+code, pre, .sourceCode { @include monospaced; }
+
+code, pre { width: min-content; }
```
Comment (ncfavier) [inline support/web/css/code.scss:17]: Uuuh

screenshot-2024-01-27-094743

Comment (plt-amy) [reply support/web/css/code.scss:17]: open a new issue

**Pull request merged**